packages/core/src/browser/opener-service.ts

```diff
@@ -94,9 +94,13 @@ export class DefaultOpenerService implements OpenerService {
     }
 
     protected async prioritize(uri: URI, options?: OpenerOptions): Promise<OpenHandler[]> {
-        const prioritized = await Prioritizeable.prioritizeAll(this.getHandlers(), handler =>
-            handler.canHandle(uri, options)
-        );
+        const prioritized = await Prioritizeable.prioritizeAll(this.getHandlers(), handler => {
+            try {
+                return handler.canHandle(uri, options);
```

Comment:
Would it make sense for other usages of `Prioritizeable`? We could catch once in `Prioritizeable.toPrioritizeable` and `Prioritizeable.toPrioritizeableSync`.

Reply:
Prioritzable could be used elsewhere, in which case the catching undesirably swallows exceptions. I put the catching here because exceptions thrown from contributions should not break the application or other contributions, no matter they are using Prioritizeable or not.